dsl/pace/dsl/stencil.py

```diff
@@ -213,65 +215,48 @@ def __init__(
             and MPI.COMM_WORLD.Get_size() > 1
             and "dace" not in self.stencil_config.backend
         ):
-            stencil_function = future_stencil.future_stencil
-            stencil_kwargs["wrapper"] = self
+            self.stencil_function = future_stencil.future_stencil
+            self.stencil_kwargs["wrapper"] = self
         else:
             # future stencil provides this information and
             # we want to be consistent with the naming whether we are
             # running in parallel or not (so we use the same cache)
-            stencil_kwargs["name"] = func.__module__ + "." + func.__name__
+            self.stencil_kwargs["name"] = self.func.__module__ + "." + self.func.__name__
 
         if skip_passes and self.stencil_config.is_gtc_backend:
-            stencil_kwargs["skip_passes"] = skip_passes
-        if "skip_passes" in stencil_kwargs:
-            stencil_kwargs["oir_pipeline"] = FrozenStencil._get_oir_pipeline(
-                stencil_kwargs.pop("skip_passes")
+            self.stencil_kwargs["skip_passes"] = skip_passes
+        if "skip_passes" in self.stencil_kwargs:
+            self.stencil_kwargs["oir_pipeline"] = FrozenStencil._get_oir_pipeline(
+                self.stencil_kwargs.pop("skip_passes")
             )
-
-        # When using DaCe orchestration, we deactivate code generation
-        # (Only SDFG are needed). But because some stencils are executed
-        # outside of the runtime path, we have a whitelist exception.
-        if (
-            dace_config.is_dace_orchestrated()
-            and not is_dacemode_codegen_whitelisted(func)
-            and "dace" in self.stencil_config.backend
-        ):
-            stencil_kwargs["disable_code_generation"] = True
-
-        self.stencil_object: gt4py.StencilObject = stencil_function(
-            definition=func,
-            externals=externals,
-            **stencil_kwargs,
-        )
-        """generated stencil object returned from gt4py."""
-
-        self._argument_names = tuple(inspect.getfullargspec(func).args)
+            
+        self._argument_names = tuple(inspect.getfullargspec(self.func).args)
 
         assert (
             len(self._argument_names) > 0
         ), "A stencil with no arguments? You may be double decorating"
 
-        field_info = self.stencil_object.field_info
-        self._field_origins: Dict[
-            str, Tuple[int, ...]
-        ] = FrozenStencil._compute_field_origins(field_info, self.origin)
-        """mapping from field names to field origins"""
+    def _compile(self):
+        if self.stencil_object:
```

Comment:
should we also check that `self.stencil_object` and `self._stencil_run_kwargs` are set?